src/main/java/net/dv8tion/jda/api/utils/SessionController.java

```diff
@@ -274,4 +302,68 @@ public int getConcurrency()
          */
         void run(boolean isLast) throws InterruptedException;
     }
+
+    /**
+     * Wrapper for {@link #getGlobalRatelimit()} and {@link #setGlobalRatelimit(long)}.
+     */
+    @Deprecated
+    @ForRemoval(deadline = "5.0.0")
+    @ReplaceWith("getRateLimitHandle()")
+    @SuppressWarnings("DeprecatedIsStillUsed")
+    class GlobalRateLimitAdapter implements RestRateLimiter.GlobalRateLimit
```

Comment:
Why is this set for deprecation and removal and setup with a ReplaceWith when it is used in the thing that is indicated to replace with? I'm confused.

Reply:
Is the hope here that developers implement their own GlobalRateLimit implementation to return from SessionController?

Reply:
I don't quite understand what this is for.

Reply:
This is for backwards compatibility. We previously handled global rate-limits using `get/setGlobalRatelimit` on the controller, which is no longer used by JDA. The reason this adapter was introduced, is to allow existing implementations to continue working for now. However, as the annotations imply, we will remove both those getters/setters and the adapter in the full release.

Most people won't be affected by this since our default implementation is already updated.